scripts/pythonscripts/sync_postgres_to_mssql.py

```diff
@@ -0,0 +1,228 @@
+        sql_cursor.execute(create_table_sql.replace(f"kawasaki.{table_name}", temp_table_name))
+        sql_conn.commit()
+
+        # Use pandas to read in chunks
+        for chunk in pd.read_sql(
+            f'SELECT * FROM curated_kawasaki_registry.{table_name}',
+            pg_conn,
+            chunksize=chunk_size
+        ):
+            chunk_start = time.time()
+            
+            # Handle datetime columns
+            datetime_columns = [col[0] for col in columns if col[1] in ['timestamp without time zone', 'datetime2']]
+            for col in datetime_columns:
+                chunk[col] = pd.to_datetime(chunk[col], errors='coerce')
+                chunk.loc[chunk[col] < '1753-01-01', col] = pd.NaT
+                chunk.loc[chunk[col] > '9999-12-31', col] = pd.NaT
+            
+            # Replace NaN/inf values with None
+            chunk = chunk.replace([np.inf, -np.inf], np.nan)
+            chunk = chunk.where(pd.notnull(chunk), None)
+            
+            # Fast batch insert into temp table
+            sql_cursor.fast_executemany = True
+            sql_cursor.executemany(insert_sql.replace(f"kawasaki.{table_name}", temp_table_name), 
+                                 chunk.values.tolist())
+            sql_conn.commit()
+            
+            # Move data from temp to final table in smaller batches
+            sql_cursor.execute(f"""
+                INSERT INTO kawasaki.{table_name}
+                SELECT TOP ({len(chunk)}) * FROM {temp_table_name};
+                DELETE TOP ({len(chunk)}) FROM {temp_table_name};
+            """)
+            sql_conn.commit()
+            
+            rows_copied += len(chunk)
+            chunk_time = time.time() - chunk_start
+            rows_per_second = len(chunk) / chunk_time if chunk_time > 0 else 0
+            
+            print(f"Progress for {table_name}: {rows_copied:,}/{total_rows:,} rows "
+                  f"({(rows_copied/total_rows*100):.1f}%) - "
+                  f"Speed: {rows_per_second:.0f} rows/sec")
+
+        # Drop temp table
+        sql_cursor.execute(f"DROP TABLE {temp_table_name}")
+        sql_conn.commit()
+
+        end_time = time.time()
+        total_time = end_time - start_time
+        avg_speed = rows_copied / total_time if total_time > 0 else 0
+        print(f"Completed copying {table_name}: {rows_copied:,} rows in {total_time:.2f} seconds "
+              f"(avg {avg_speed:.0f} rows/sec)")
+        
+        # Verify row count
+        sql_cursor.execute(f"SELECT COUNT(*) FROM kawasaki.{table_name}")
+        final_count = sql_cursor.fetchone()[0]
+        print(f"Final row count in destination table {table_name}: {final_count:,}")
+        if final_count != total_rows:
+            print(f"WARNING: Row count mismatch for {table_name}! Source: {total_rows:,}, Destination: {final_count:,}")
+
+    except Exception as e:
+        print(f"Error copying table {table_name}: {str(e)}")
```

Comment:
if there is an error, consider raising an exception:
```
raise RuntimeError("ERROR HERE")
```
List of errors: https://docs.python.org/3/library/exceptions.html